lib/aws-s3-stack.ts

```diff
@@ -1,8 +1,56 @@
 import * as cdk from 'aws-cdk-lib';
 import { Construct } from 'constructs';
+import { AwsS3StackProps } from './AwsS3StackProps';
+import * as s3 from 'aws-cdk-lib/aws-s3';
 
 export class AwsS3Stack extends cdk.Stack {
-  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+  /**
+   * Constructs a new instance of the AwsS3Stack.
+   *
+   * @param scope The parent construct.
+   * @param id An identifier for the stack.
+   * @param props Configuration properties for the stack, including resource prefixes, bucket names, and deployment environment.
+   */
+  constructor(scope: Construct, id: string, props: AwsS3StackProps) {
     super(scope, id, props);
+
+    const removalPolicy = props.deployEnvironment === 'production' ? cdk.RemovalPolicy.RETAIN : cdk.RemovalPolicy.DESTROY;
+
+    // define an S3 bucket
+    const s3Bucket = new s3.Bucket(this, `${props.resourcePrefix}-${props.s3BucketName}`, {
+      bucketName: `${props.deployEnvironment}-${props.deployRegion}-${props.s3BucketName}`,
+      encryption: s3.BucketEncryption.S3_MANAGED,
```

Comment:
**Suggestion:** Implement server-side encryption using AWS KMS instead of S3-managed keys for enhanced security and audit capabilities, allowing better control over encryption keys.
> 🚢  **High Priority** 
 >[Category: **`Security`**, Importance: **`10`**]
> **Reason:** Switching to KMS-managed keys for server-side encryption significantly enhances security and provides better control and audit capabilities for encryption keys. This is a critical improvement for ensuring data security and compliance.

```suggestion
      encryption: s3.BucketEncryption.KMS_MANAGED,
      encryptionKey: new kms.Key(this, `${props.resourcePrefix}-bucket-key`, {
        enableKeyRotation: true,
        description: `KMS key for ${props.s3BucketName} bucket encryption`,
        alias: `${props.resourcePrefix}-bucket-key`,
      }),
```
- [ ]  **I have reviewed this code suggestion and addressed the relevant ones.**